civil_society_vote/civil_society_vote/middleware.py

```diff
@@ -6,3 +8,19 @@ def middleware(request):
         return response
 
     return middleware
+
+
+class CaseInsensitiveUserModel(object):
+    def authenticate(self, request, username=None, password=None):
```

Comment:
While this approach might work now as a quick fix, I feel like we are hard-coding some of the inner workings and conventions of the framework with unintended side effects. 

Because we don't have tests covering the auth and permissions at this time and we cannot guarantee that this does not affect other things, I would recommend a more conservative approach. Namely, to lowercase the email on registration and profile save (here: https://github.com/code4romania/votong/blob/develop/civil_society_vote/accounts/models.py#L9).

```python 
def save(self, *args, **kwargs):
    self.username = self.username.lower()
    self.email = self.email.lower()
    super().save(*args, **kwargs)
```

For the existing accounts, a migration script or management command can be made, or even simpler just open a management shell on the prod machine and run:
```python
$ ./manage.py shell
>>> from accounts.models import User
>>> for user in User.objects.all():
...    user.username =  user.username.lower()
...    user.email =  user.email.lower()
...    user.save()
``` 
WDYT?

Reply:
Just realized that we have to account to users having already saved the upper cased emails in their password manager so we would need to lowercase the entries on the login as well. So the middleware approach is still needed. 😓

Reply:
What if we subclass `CaseInsensitiveUserModel` from `ModelBackend`? Then do something like:
```python
def authenticate(self, request, username=None, password=None, **kwargs):
    if username:
        username = username.lower()
    super().authenticate(request, username, password, **kwargs)
```
Still a bit hacky, but it should work just as well. Let me know your thougths.

Reply:
Python and Django are not my main specialty, so I might be wrong on this, but as far as I understand it, the middleware that I added is the fallback in case [all other authentication backends](https://github.com/code4romania/votong/pull/227/files#diff-2d7055d8a881fecdc7905093a28a44fd04f9390cfe34e9aa67e575e4471cef89R80-R81) have failed to produce as successful login. 

Nothing is altered in the DB, and if anybody could have logged-in before should be able to continue to do so.

Reply:
Yeah, you are correct. Might be overthinking things here. 😅 Hopefully we'll manage to get some tests written for this soon and have more confidence with other upcoming changes.

Thanks for your contribution. 👍

Reply:
My pleasure. Glad it helps 😁